OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -210,22 +207,26 @@ public void redoAction() {
     }
 
     public File getCurrentFile() {
-        return currentFile;
+        return editorState.getStateCurrentFile();
     }
 
     public void setCurrentFile(File currentFile) {
-        this.currentFile = currentFile;
+        editorState.setStateCurrentFile(currentFile);
         setDisplayName(currentFile.getName());
     }
 
     public DbxState getDbxState() {
-        return dbxState;
+        return editorState.getStateDbxState();
     }
 
     public void setDbxState(DbxState dbxState) {
```

Comment:
All of the above getters/setters should be removed. Only the editorState getter/setter should be exposed and any other class that wants to modify the state should go through editor state getter/setter